src/Native/Signal.js

```diff
@@ -118,11 +118,19 @@ Elm.Native.Signal.make = function(localRuntime) {
     input.kids.push(this);
   }
 
-  function timestamp(a) {
-    function update() { return Utils.Tuple2(localRuntime.timer.now(), a.value); }
-    return new LiftN(update, [a]);
+  function Timestamp(input) {
+    this.id = Utils.guid();
+    this.value = Utils.Tuple2(localRuntime.timer.programStart, input.value);
+    this.kids = [];
+    this.recv = function(timestep, changed, parentID) {
+      if (changed) { this.value = Utils.Tuple2(timestep, input.value); }
```

Comment:
Use more modern style. I did a bunch of messy things when I was writing this code.

```javascript
if (changed) {
  this.value = Utils.Tuple2(timestep, input.value);
}
```